modules/libarchive/3.7.3/patches/build_with_bazel.patch

```diff
@@ -0,0 +1,1805 @@
++#define HAVE_DECL_UINT64_MAX 1
++#define HAVE_DECL_UINTMAX_MAX 1
++#define HAVE_DIRECT_H 1
++#define HAVE_EILSEQ 1
++#define HAVE_ERRNO_H 1
++#define HAVE_FCNTL_H 1
++#define HAVE_FSTAT 1
++#define HAVE_GETPID 1
++#define HAVE_INTTYPES_H 1
++#define HAVE_IO_H 1
++#define HAVE_LIMITS_H 1
++#define HAVE_LOCALE_H 1
++#define HAVE_MBRTOWC 1
++#define HAVE_MEMMOVE 1
++#define HAVE_MEMORY_H 1
++#define HAVE_MKDIR 1
++#define HAVE_PROCESS_H 1
++#define HAVE_SETLOCALE 1
++#define HAVE_SIGNAL_H 1
++#define HAVE_STDARG_H 1
++#define HAVE_STDINT_H 1
++#define HAVE_STDLIB_H 1
++#define HAVE_STRCHR 1
++#define HAVE_STRNLEN 1
++#define HAVE_STRDUP 1
++#define HAVE_STRERROR 1
++#define HAVE_STRFTIME 1
++#define HAVE_STRING_H 1
++#define HAVE_STRRCHR 1
++#define HAVE_SYS_STAT_H 1
++#define HAVE_SYS_TYPES_H 1
++#define HAVE_SYS_UTIME_H 1
++#define HAVE_TIME_H 1
++#define HAVE_TZSET 1
++#define HAVE_UTIME 1
++#define HAVE_WCHAR_H 1
++#define HAVE_WCHAR_T 1
++#define HAVE_WCRTOMB 1
++#define HAVE_WCSCMP 1
++#define HAVE_WCSCPY 1
++#define HAVE_WCSLEN 1
++#define HAVE_WCTOMB 1
++#define HAVE_WCTYPE_H 1
++#define HAVE_WINCRYPT_H 1
++#define HAVE_WINDOWS_H 1
++#define HAVE_WINIOCTL_H 1
++#define HAVE__CrtSetReportMode 1
++#define HAVE_CTIME_S 1
++#define HAVE__FSEEKI64 1
++#define HAVE__GET_TIMEZONE 1
++#define HAVE_GMTIME_S 1
++#define HAVE_LOCALTIME_S 1
++#define HAVE__MKGMTIME 1
++#define ICONV_CONST
++#define HAVE_LZ4HC_H 1
++
++#define LIBARCHIVE_VERSION_NUMBER "3007003"
++#define LIBARCHIVE_VERSION_STRING "3.7.3"
++#define SIZEOF_WCHAR_T 2
++#define VERSION "3.7.3"
++
++#define gid_t short
++#define id_t short
++#define mode_t unsigned short
++#define pid_t int
++#define ssize_t int64_t
++#define uid_t short
+diff --git a/libarchive_fe/BUILD.bazel b/libarchive_fe/BUILD.bazel
+new file mode 100644
+index 00000000..8776c21c
+--- /dev/null
++++ b/libarchive_fe/BUILD.bazel
+@@ -0,0 +1,18 @@
++load("@rules_cc//cc:defs.bzl", "cc_library")
++load("//bazel:config.bzl", "PLATFORM_LOCAL_DEFINES")
++
++package(default_visibility = ["//visibility:public"])
++
++cc_library(
++    name = "archive_fe",
++    hdrs = glob(["*.h"]),
++    srcs = glob(["*.c"]),
++    defines = PLATFORM_LOCAL_DEFINES,
++    deps = ["//:config"],
++    strip_include_prefix = ".",
++)
++
++alias(
++    name = "libarchive_fe",
++    actual = ":archive_fe",
++)
+diff --git a/tar/BUILD.bazel b/tar/BUILD.bazel
+new file mode 100644
+index 00000000..f326ffd9
+--- /dev/null
++++ b/tar/BUILD.bazel
+@@ -0,0 +1,40 @@
++load("@rules_cc//cc:defs.bzl", "cc_binary")
++
++package(default_visibility = ["//visibility:public"])
++
++SRCS = [
++    "bsdtar.c",
++    "bsdtar.h",
++    "bsdtar_platform.h",
++    "cmdline.c",
++    "creation_set.c",
++    "read.c",
++    "subst.c",
++    "util.c",
++    "write.c",
++]
++
++cc_binary(
++    name = "bsdtar",
++    target_compatible_with = select({
++        # TODO: not working on macos
++        "@platforms//os:macos": ["@platforms//:incompatible"],
```

Comment:
hmmm, this was working on macos before. What's not working now?
I think this is a blocker for landing

Reply:
I re-enabled it on my fork to remind myself: https://github.com/zaucy/libarchive/actions/runs/8953745198/job/24592569883

```
tar/bsdtar.c:1078:50: error: expected ';' after top level declarator
        "  -f <filename>  Location of archive (default " _PATH_DEFTAPE ")\n"
```

I don't have a mac so I wasn't sure what was wrong with it. Works on Windows and Ubuntu on my machine and github actions.

Reply:
ah, that's right. I remember looking at https://github.com/libarchive/libarchive/blob/6818dd167cb190c76e0cfdeb40cd73aa22b96bd8/tar/bsdtar.c#L72-L83 and being a little confused how it's supposed to work on a mac.

It seems like not having it defined is fine, it just [requires patching the help message](https://github.com/bazelbuild/bazel-central-registry/blob/fcb922a1fa835db5bc1199d591c54b02dc73e2b3/modules/libarchive/3.7.2.bcr.2/patches/build_with_bazel.patch#L332-L333)

Reply:
ah! thanks. That is a little weird that it isn't an issue for them upstream 🤔